core/node/proof_data_handler/src/lib.rs

```diff
@@ -88,17 +96,23 @@ fn create_proof_processing_router(
 
     if config.tee_support {
         let get_tee_proof_gen_processor =
-            TeeRequestProcessor::new(blob_store, connection_pool, config.clone());
+            TeeRequestProcessor::new(blob_store, connection_pool, config.clone(), l2_chain_id);
         let submit_tee_proof_processor = get_tee_proof_gen_processor.clone();
         let register_tee_attestation_processor = get_tee_proof_gen_processor.clone();
 
         router = router.route(
             "/tee/proof_inputs",
             post(
                 move |payload: Json<TeeProofGenerationDataRequest>| async move {
-                    get_tee_proof_gen_processor
+                    let result: Result<Json<zksync_prover_interface::api::TeeProofGenerationDataResponse>, errors::RequestProcessorError> = get_tee_proof_gen_processor
                         .get_proof_generation_data(payload)
-                        .await
+                        .await;
+
+                    match result {
+                        Ok(Json(TeeProofGenerationDataResponse(None))) => (StatusCode::NO_CONTENT, Json("No new TeeVerifierInputs are available yet")).into_response(),
```

Comment:
"No content" implies empty response body; not sure whether it'll be correctly handled by clients though (but a string response would probably require special handling as well).

Reply:
@pbeza ?

Reply:
Great catch! I forgot to test the client after making this change.

I've addressed it in this PR: https://github.com/matter-labs/zksync-era/pull/3067 and have re-tested it manually end-to-end. Please merge it if you approve.

Reply:
@slowli Would you mind reviewing #3067, which addresses your comment?

I’ve resolved the issue, but now I'm returning a `StatusCode::NOT_FOUND` (`404`) HTTP status code instead of `StatusCode::NO_CONTENT` (`204`). This is because we already have a mapping from `zksync_dal::SqlError::RowNotFound` to `StatusCode::NOT_FOUND` here:

https://github.com/matter-labs/zksync-era/blob/748ee1b831c5fc81f6df2442dab3c862f5e19462/core/node/proof_data_handler/src/errors.rs#L32-L34

and I'm now using:

https://github.com/matter-labs/zksync-era/blob/ab6e56552377ea74ccf619f940a03735c0e4d5ea/core/lib/dal/src/tee_proof_generation_dal.rs#L108

(that throws `zksync_dal::SqlError::RowNotFound`), instead of:

https://github.com/matter-labs/zksync-era/blob/3d77b90315a67ce361fc86315021ec0c403d0cbb/core/lib/dal/src/tee_proof_generation_dal.rs#L108

@haraldh suggested that we should return `StatusCode::NO_CONTENT` if no new `TeeVerifierInput` is available yet. However, I don't see an easy way to bypass this existing mapping unless we modify it globally to return `StatusCode::NO_CONTENT` instead of `StatusCode::NOT_FOUND`.

Reply:
I've prepared an alternative solution in https://github.com/matter-labs/zksync-era/pull/3082. Feel free to choose whichever you prefer, or suggest another solution that I can implement.

Reply:
@slowli, could you please review? We would like to get this PR wrapped up and merged so we can move on to other tasks that depend on it. It's a big one that affects a lot, so I would really prefer to have it merged before we move forward.

Reply:
I like "not found" errors better; they look more idiomatic.

Reply:
@haraldh, the `not found` error is from PR #3067. Are you fine with merging it?

Reply:
merged with changes